bin/resin-preload

```diff
@@ -203,33 +202,16 @@ getResinSdk(options)
       reject(err)
     })
 
-    return preloader.build()
-    .then(() => {
-      return preloader.prepare()
-    })
-    .then(() => {
-      return preloader.fetchApplication()
-    })
+    return preloader.prepare()
     .then(() => {
-      if (preloader.application.build.length === 0) {
-        reject(new ExitError('This application has no successful builds'))
-      }
-      return preloader.getDeviceTypeAndPreloadedBuilds()
-    })
-    .then((info) => {
-      if (options.dontCheckDeviceType === false && info.device_type !== preloader.application.device_type) {
-        const msg = `Application device type (${preloader.application.device_type}) and image device type (${info.device_type}) do not match.`
+      const msg = preloader.cantPreload()
```

Comment:
`cantPreload` is a slightly weird name for a function that returns an error message if some conditions aren't met. What this actually does is validate that preloading can be performed, so in my opinion it should named something like `ensureCanPreload` that *throws errors* if any of the conditions it checks aren't met.

Reply:
I didn't want this function to throw to simplify the resin-preload api.

Reply:
Finally I updated this to be _ensureCanPreload: a method that throws a ResinError if something is wrong.